qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java

```diff
@@ -0,0 +1,131 @@
+
+    @Test
+    public void testHttpGetRequestWithStringMapping() {
+        HTTPDao httpDao = new HTTPDao(HOST);
+        String result = httpDao.getAsString("/comments/1", MediaType.APPLICATION_JSON);
+        Assert.assertEquals(result, """
+                {
+                  "postId": 1,
+                  "id": 1,
+                  "name": "id labore ex et quam laborum",
+                  "email": "Eliseo@gardner.biz",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""");
+    }
+
+    @Test
+    public void testHttpGetRequestWithResponseObject() {
+        HTTPDao httpDao = new HTTPDao(HOST);
+        Response result = httpDao.get("/comments/1", MediaType.APPLICATION_JSON);
+        Assert.assertEquals(result.getStatus(), 200);
+        String json = result.readEntity(String.class);
+        Assert.assertEquals(json, """
+                {
+                  "postId": 1,
+                  "id": 1,
+                  "name": "id labore ex et quam laborum",
+                  "email": "Eliseo@gardner.biz",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""");
+    }
+
+    @Test
+    public void testHttpPostRequestWithStringMapping() {
+        String newCommentEntity = """
+                {
+                  "postId": 1,
+                  "id": 501,
+                  "name": "id labore ex et quam laborum",
+                  "email": "Eliseo@gardner.biz",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""";
+        HTTPDao httpDao = new HTTPDao(HOST);
+        String result = httpDao.postAsString("/comments", MediaType.APPLICATION_JSON, newCommentEntity);
+        Assert.assertEquals(result, newCommentEntity);
+    }
+
+    @Test
+    public void testHttpPostRequestWithResponseObject() {
+        HTTPDao httpDao = new HTTPDao(HOST);
+        String newCommentEntity = """
+                {
+                  "postId": 1,
+                  "id": 501,
+                  "name": "id labore ex et quam laborum",
+                  "email": "Eliseo@gardner.biz",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""";
+        try (Response result = httpDao.post("/comments", MediaType.APPLICATION_JSON, newCommentEntity)) {
+            Assert.assertEquals(result.getStatus(), 201);
+            String json = result.readEntity(String.class);
+            Assert.assertEquals(json, newCommentEntity);
+        }
+    }
+
+    @Test
```

Comment:
Please add a comment to the test case

Reply:
I am not the author of this test case, I'm not quite sure what to properly comment on.